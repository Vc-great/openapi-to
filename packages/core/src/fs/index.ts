export { clean } from './clean.ts'
export { pathExistsSync } from './pathExistsSync.ts'
export { read, readSync } from './read.ts'
export { write } from './write.ts'
