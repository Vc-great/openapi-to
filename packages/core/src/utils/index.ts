export * from "./cache.ts";
export * from "./cache.ts";
export * from "./clean.ts";
export * from "./logger.ts";
export * from "./pathExistsSync.ts";
export * from "./promise.ts";
export * from "./Queue.ts";
export * from "./Queue.ts";
export * from "./randomColour.ts";
export * from "./read.ts";
export * from "./renderTemplate.ts";
export * from "./throttle.ts";
export * from "./timeout.ts";
export * from "./transformers/index.ts";
export * from "./uniqueName.ts";
export * from "./URLPath.ts";
export * from "./write.ts";
