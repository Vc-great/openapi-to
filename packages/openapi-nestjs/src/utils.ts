import stringifyObject from "./stringifyObject.ts";

export function formatterType(type: string): string {
  if (type === "integer") {
    return "number";
  }
  return type;
}

export function objectToStringify(object: unknown): string {
  return stringifyObject(object, {
    filter(obj: { [k: string]: unknown }, key: string) {
      return obj[key] !== undefined;
    },
  });
}
