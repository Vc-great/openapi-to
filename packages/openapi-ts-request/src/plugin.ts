import {AST,OpenAPI,createPlugin} from '@openapi-to/core'
import {RequestGenerator} from "./RequestGenerator.ts";
import Oas from 'oas';
import type {PluginConfig} from './types.ts'

export const definePlugin = createPlugin<PluginConfig>((pluginConfig)=>({openapiDocument,openapiToSingleConfig}) => {
  const ast = new AST()
  const oas = new Oas({...openapiDocument})
  const openapi = new OpenAPI({}, oas);
  return {
    name:"openapi-ts-request",
   buildStart(){
     const requestGenerator = new RequestGenerator({
       oas,
       ast,
       openapi,
       pluginConfig,
       openapiToSingleConfig,
     });
     requestGenerator.build()
    },
    writeFile(){
      ast.saveSync()
    },
    buildEnd(){
      //log
    }
  }

})
