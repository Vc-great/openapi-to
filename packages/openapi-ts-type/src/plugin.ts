import { AST, createPlugin, OpenAPI } from "@openapi-to/core";

import Oas from "oas";

import { TypeGenerator } from "./TypeGenerator.ts";

import type { PluginConfig } from "./types.ts";

export const definePlugin = createPlugin(
  (pluginConfig?: PluginConfig) =>
    ({ openapiDocument, openapiToSingleConfig }) => {
      const ast = new AST();
      const oas = new Oas({ ...openapiDocument });
      const openapi = new OpenAPI({}, oas);
      return {
        name: "openapi-ts-type",
        buildStart(context) {
          const requestGenerator = new TypeGenerator({
            oas,
            ast,
            openapi,
            pluginConfig,
            openapiToSingleConfig,
          });
          requestGenerator.build(context);
        },
        writeFile() {
          return ast.saveSync();
        },
        buildEnd() {},
      };
    },
);
