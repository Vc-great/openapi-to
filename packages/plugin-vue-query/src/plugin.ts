import { AST, createPlugin, OpenAPI, pluginEnum } from "@openapi-to/core";

import Oas from "oas";

import { Generator } from "./Generator.ts";

import type { PluginConfig } from "./types.ts";

export const definePlugin = createPlugin<PluginConfig>(
  (pluginConfig) =>
    ({ openapiDocument, openapiToSingleConfig }) => {
      const ast = new AST();
      const oas = new Oas({ ...openapiDocument });
      const openapi = new OpenAPI({}, oas);
      return {
        name: pluginEnum.VueQuery,
        buildStart(context) {
          const requestGenerator = new Generator({
            oas,
            ast,
            openapi,
            pluginConfig,
            openapiToSingleConfig,
          });
          requestGenerator.build(context);
        },
        writeFile() {
          return ast.sourceFile.map((item) => {
            return {
              filePath: item.getFilePath(),
              fileText: item.getFullText(),
            };
          });
        },
        buildEnd() {},
      };
    },
);
