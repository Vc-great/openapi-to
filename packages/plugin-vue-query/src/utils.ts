import _ from 'lodash'

import { VUE_QUERY_SUFFIX } from './constants.ts'

export function formatTypeNamespaceName(typeName: string): string {
  return _.upperFirst(typeName) + _.upperFirst(VUE_QUERY_SUFFIX)
}

export function formatQueryName(name: string): string {
  return `${_.lowerFirst(name)}${_.upperFirst(VUE_QUERY_SUFFIX)}`
}

export function formatQueryFileName(name: string): string {
  return `${_.lowerFirst(name)}.${VUE_QUERY_SUFFIX}`
}
