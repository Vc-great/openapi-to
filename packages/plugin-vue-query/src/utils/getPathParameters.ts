import type {OperationWrapper} from "@openapi-to/core";
import {camelCase} from "lodash-es";
import {OpenAPIV3} from "openapi-types";
import type {OptionalKind, ParameterDeclarationStructure} from "ts-morph";

export  function getPathParameters(operation: OperationWrapper) {
  const pathParameters: OptionalKind<ParameterDeclarationStructure>[] = operation.accessor.pathParameters.map((item: OpenAPIV3.ParameterObject) => {
    return {
      name: camelCase(item.name),
      type: `MaybeRefOrGetter<${operation.accessor.operationTSType?.pathParams || ''}['${camelCase(item.name)}']>`,
    }
  })
 return pathParameters
}
