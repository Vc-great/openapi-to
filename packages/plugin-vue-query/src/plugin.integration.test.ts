//@ts-nocheck
import fs from 'node:fs'
import path from 'node:path'
import { fileURLToPath } from 'node:url'
import { PluginManager, pluginEnum } from '@openapi-to/core'
import { checkFolderHasFiles } from '@openapi-to/core/utils'

import { afterEach, beforeEach, describe, expect, it } from 'vitest'
import mockOpenAPI from '../mock/petstore.json'
import { definePlugin } from './plugin'

// 导入TsType插件
import { definePlugin as defineTsTypePlugin } from '@openapi-to/plugin-ts-type'
import { definePlugin as defineTsRequestPlugin } from '@openapi-to/plugin-ts-request'

// 获取当前文件的目录
const __dirname = path.dirname(fileURLToPath(import.meta.url))
const TEST_OUTPUT_DIR = path.resolve(__dirname, '../test-output')

describe('Ts Request Plugin Integration', () => {
  beforeEach(() => {
    // 清理并重新创建测试输出目录
    if (fs.existsSync(TEST_OUTPUT_DIR)) {
      fs.rmSync(TEST_OUTPUT_DIR, { recursive: true, force: true })
    }
    fs.mkdirSync(TEST_OUTPUT_DIR, { recursive: true })

    console.log(`测试目录已准备: ${TEST_OUTPUT_DIR}`)
  })

  afterEach(() => {
    // 确保测试目录存在
    if (!fs.existsSync(TEST_OUTPUT_DIR)) {
      fs.mkdirSync(TEST_OUTPUT_DIR, { recursive: true })
    }
  })

  it('应该生成默认的服务类文件', async () => {
    const pluginManager = new PluginManager(
      {
        name: 'request',
        root: '',
        plugins: [defineTsTypePlugin(), defineTsRequestPlugin(),definePlugin()],
        input: {
          path: '',
        },
        output: {
          dir: TEST_OUTPUT_DIR,
        },
      },
      // @ts-ignore
      mockOpenAPI,
    )

    await pluginManager.run()

    // 检查文件是否生成
    const hasFiles = checkFolderHasFiles(TEST_OUTPUT_DIR)
    expect(hasFiles).toBe(true)

    // 修正为正确的文件路径 - 检查pet目录
    const petDir = path.join(TEST_OUTPUT_DIR, 'pet')
    expect(fs.existsSync(petDir)).toBe(true)

    // 查找pet目录下的所有service文件
    const serviceFiles = fs.readdirSync(petDir).filter((file) => file.endsWith('.service.ts'))
    expect(serviceFiles.length).toBeGreaterThan(0)

    // 选取第一个服务文件进行内容检查
    const petServicePath = path.join(petDir, serviceFiles[0])
    const fileContent = fs.readFileSync(petServicePath, 'utf-8')

    expect(fileContent).toContain('Service')
    expect(fileContent).toMatch(/addPet|updatePet|findPetsByStatus/)

    expect(fileContent).toMatchSnapshot()
  })
})
