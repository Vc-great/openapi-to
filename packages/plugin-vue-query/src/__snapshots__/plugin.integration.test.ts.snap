// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Ts Request Plugin Integration > 应该生成默认的服务类文件 1`] = `
"import type { AxiosRequestConfig } from "axios";
import { toValue } from "vue";
import type { MaybeRefOrGetter } from "vue";
import { useMutation } from "@tanstack/vue-query";
import type { UseMutationOptions } from "@tanstack/vue-query";
import type { AddPetMutationRequest, AddPetMutationResponse, AddPetResponseError } from "./add-pet.types.ts";
import { addPetService } from "./add-pet.service.ts";

/** the final transformed data type after \`select\` (or other transforms); this is what components receive. */
type TData = AddPetMutationResponse;

export interface TVariables {
  /** The data to be sent in the request body. */
  data: MaybeRefOrGetter<AddPetMutationRequest>;
}

export const addPetMutationKey = () => [{ url: '/pet', method: 'post' }] as const;

export type AddPetMutationKey = ReturnType<typeof addPetMutationKey>;

/**
 * @summary Add a new pet to the store
 */
export function useAddPet<TContext>(options?: {
  requestConfig?: Partial<AxiosRequestConfig<AddPetMutationRequest>>;
  mutation?: UseMutationOptions<
    AddPetMutationResponse,
    AddPetResponseError,
    TVariables,
    TContext
  >;
}) {

  const { mutation: mutationOptions = {}, requestConfig = {} } = options ?? {};
  const mutationKey = addPetMutationKey();

  return useMutation<
    TData,
    AddPetResponseError,
    TVariables,
    TContext
  >({
    mutationKey,
    mutationFn: (data) => {
      return addPetService(toValue(data), requestConfig);
    },
    ...mutationOptions
  });
}
"
`;
